sympy/vector/parametricregion.py

```diff
@@ -0,0 +1,48 @@
+from sympy.core.basic import Basic
+
+class ParametricRegion(Basic):
+    """
+    Represents a parametric region in space.
+    """
+    def __init__(self, system, *args):
+        """
+        Examples
+        ========
+
+        >>> from sympy import symbols, cos, sin, pi
+        >>> from sympy.vector import CoordSys3D, ParametricRegion
+        >>> r, theta = symbols("r theta")
+        >>> C = CoordSys3D('C')
+        >>> circle = ParametricRegion(C, r*cos(theta), r*sin(theta), (theta, 0, 2*pi))
+        """
+        self._definition = list()
+        self._bounds = list()
+        self._parameters = list()
+        self.system = system
+
+        for arg in args:
+            if isinstance(arg, tuple):
+                self.bounds.append(arg)
+            else:
+                self.definition.append(arg)
+
+        if len(self.definition) > 3:
+            raise ValueError("Only 3-D regions are supported")
```

Comment:
Why this restriction?

Reply:
I think the vector module supports only 3-D Coordinate frame. If we cannot define a vector field of more dimensions, then why allow defining a higher dimension.

Reply:
I don't see the point of having this restriction. ParametricRegion shouldn't only be defined in 3D.

Reply:
OK, I have removed these conditions in the new commit.